core/new-gui/src/app/workspace/service/workflow-collab/workflow-collab.service.ts

```diff
@@ -0,0 +1,69 @@
+import { Injectable } from "@angular/core";
+import { webSocket } from "rxjs/webSocket";
+import { CommandMessage } from "../workflow-graph/model/workflow-action.service";
+import { environment } from "../../../../environments/environment";
+import { Subject, Observable } from "rxjs";
+
+@Injectable({
+  providedIn: "root",
+})
+
+// This service will be for workflow collab
+// First step: move the arrays from undo/redo service into Yjs doc.
+// Once we have some stuff set I'll send the changes between clients and just see how to work with them.
+// Need to find a way to push as part of change, not update like that
+
+// How do we recover if the connection gets dropped?
+export class WorkflowCollabService {
+  private sendData: boolean = false; // set initial value to false to disable service
+
+  private socket = webSocket({
+    url: this.getWorkflowWebsocketUrl(),
+    deserializer: msg => msg["data"],
+  });
+
+  // maybe create a subject for testing?
+  private messageSubject: Subject<CommandMessage> = new Subject<CommandMessage>();
+
+  constructor() {
+    const self = this;
+    if (environment.enableWorkflowCollab) {
+      // to disable service, change to true to enable
+      // Noticed a strange interaction between this feature and SaveWorkflowService, things will get duplicated
```

Comment:
can you be more specific on the strange inteaction? In general we should try to solve these problems

Reply:
Seems `SaveWorkflowService` no longer exists, so this strange interaction won't appear. Deleted the comment.